app/Http/Controllers/DoctorReviewController.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Doctor;
+use App\Services\DoctorService;
+use App\Transformers\IndexDoctorReviewsResource;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class DoctorReviewController extends Controller
+{
+    public $service;
+
+    public function __construct(DoctorService $service)
+    {
+        $this->service = $service;
+    }
+
+    public function index()
+    {
+        $reviews = $this->service->getReviews();
+        return response()->json(['reviews' => IndexDoctorReviewsResource::collection($reviews), 'averageRank'=> $reviews->avg('rank')], Response::HTTP_OK);
```

Comment:
Instead of calculating the average reviews of the doctor every time we retrieve him, make a column in the doctor named "average_review" and calculate his avg every time a new review created "in an observer". And retrieve the avg from this column